lib/modules/ai_bot/commands/search_command.rb

```diff
@@ -120,12 +124,35 @@ def process(**search_args)
         )
 
       # let's be frugal with tokens, 50 results is too much and stuff gets cut off
-      limit ||= 20
-      limit = 20 if limit > 20
+      limit ||= MAX_RESULTS
+      limit = MAX_RESULTS if limit > MAX_RESULTS
+
+      should_try_semantic_search = SiteSetting.ai_embeddings_semantic_search_enabled
+      should_try_semantic_search &&= (limit == MAX_RESULTS)
+      should_try_semantic_search &&= (search_args.keys - %i[search_query order]).length == 0
```

Comment:
nit, could be `(search_args.keys - %i[search_query order]).zero?`

Reply:
true ... I so rarely use zero ActiveSupport is so magical